From 0280e921bd21d9410677d88431a170e47b94dcb5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20K=C4=99dziora?= <lkedziora@antmicro.com>
Date: Tue, 31 Mar 2026 09:15:37 +0200
Subject: [PATCH 37/39] subsys: instrumentation: Exclude irq lock/unlock and
 timestamp getters
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Having tracing and instrumentation enabled at the same time causes
tracing events to have an end timestamp earlier than the instrumented
methods that were sources of these events. This works around this fact
by avoiding emitting unnecessary events during instrumentation of
tracing event handling.

Signed-off-by: Łukasz Kędziora <lkedziora@antmicro.com>
---
 subsys/instrumentation/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/subsys/instrumentation/Kconfig b/subsys/instrumentation/Kconfig
index 0309c7b93bf..553509703c5 100644
--- a/subsys/instrumentation/Kconfig
+++ b/subsys/instrumentation/Kconfig
@@ -104,6 +104,7 @@ config INSTRUMENTATION_DYNAMIC_TRIGGER
 
 config INSTRUMENTATION_EXCLUDE_FUNCTION_LIST
 	string "Exclude function list"
+	default "arch_timing_counter_get,arch_irq_lock,arch_irq_unlock,k_cycle_get_64,sys_clock_cycle_get_64,elapsed"
 	depends on INSTRUMENTATION_MODE_CALLGRAPH || INSTRUMENTATION_MODE_STATISTICAL
 	help
 	  Set the list of function names to be excluded from instrumentation.
-- 
2.47.3

